flee/flee.py

```diff
@@ -244,7 +239,8 @@ def chooseFromWeights(self, weights, linklist) -> int:
             return -1
 
         weights = self.normalizeWeights(weights=weights)
-        return int(np.random.choice(list(range(0, len(linklist))), p=weights))
+        result = random.choices(linklist, weights=weights)
+        return result[0]
```

Comment:
np.random.choice was very slow, partly because it is being run over a list. random.choices was a much faster option that also supported weights. 
We also returned the location rather than an index.